utils/tmp_files.py

```diff
@@ -0,0 +1,40 @@
+import os
+import tempfile
+
+def setup_temp_dir(agentConfig, proc_name=None):
+    # if it has neither of these, then just drop back
+    if not agentConfig["use_dd_temp_dir"] and not agentConfig["custom_temp_dir"]:
+        return
+
+    if agentConfig["custom_temp_dir"]:
+        temp_dir = agentConfig["custom_temp_dir"]
+
+    if agentConfig["use_dd_temp_dir"]:
+        temp_dir = os.path.join(__file__, '..', 'tmp')
```

Comment:
Maybe you meant `os.path.join(os.path.basename(__file__), '..', '..', 'tmp')` ? (since this was in util.py previously?)